core/annotations.h

```diff
@@ -48,7 +48,7 @@
  * @brief Annotation handler registration routines.
  */
 
-#ifdef CLIENT_INTERFACE
+# ifdef CLIENT_INTERFACE
```

Comment:
I don't think we want to indent annotations.{c,h} for the ANNOTATIONS define -- A) it's around the entire file and B) it should be removed at some point: it was there as a way to disable annotations during development.  Better to avoid code churn when it's removed.

These should either be excluded from this rule, or there should be some general mechanism to not indent for some entire-file ifdefs.